.github/workflows/rust.yml

```diff
@@ -41,6 +50,14 @@ jobs:
           toolchain: stable
           override: true
 
+      - name: Cache Rust dependencies download
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-rust-deps
+        with:
+          path: ~/.cargo/registry
+          key: ${{ runner.os }}-download-${{ env.cache-name }}-${{ hashFiles('**/Cargo.lock' }}
```

Comment:
why is this different from above?

Also, we had mixed success with caching in general. I think having it keyed by `Cargo.lock` is a great idea! 

Not sure about caching `target` though. That was usually a performance regression, especially since it just grows without bounds.

Reply:
Probably having it keyed by `Cargo.lock` is a good compromise. Ideally, I'd prefer the caches to be persistent even across changes, since the bump of a single dependency would otherwise trigger a full rebuild. That said, we can look into [`cargo-sweep`](https://github.com/holmgr/cargo-sweep) to keep the caches small.

Reply:
That also goes on the assumption that caches are significantly faster than rebuilding. I remember that this is generally the case in symbolicator as long as:
- we make sure that they do not keep growing
- GHA has faster caches than Travis

Reply:
Even better, we can also disable incremental compilation. See https://github.com/rust-lang/cargo/pull/4817

Reply:
Keying on cargo.lock was indeed a compromise between caches growing large without doing too much or relying on smart stuff that could go wrong.  It's also why I decided to key on rustc.  Currently a full build is about 6mins on CI with the cached version 3mins (very rough approximations).

`cargo-sweep` looks interesting, but for the compilation times we have currently I'm tempted to say the complexity isn't worth it.  We could always add it later.

Reply:
> Even better, we can also disable incremental compilation. See [rust-lang/cargo#4817](https://github.com/rust-lang/cargo/pull/4817)

I guess this speeds up compilation?  Could try it out.

Reply:
It speeds up repeated compilation when you only change single files in a crate. For CI, I think it matters less, and at least it would get rid of most of the bloat in `target`.